PocketKit/Sources/PocketKit/Collections/CollectionViewController.swift

```diff
@@ -318,6 +289,123 @@ private extension CollectionViewController {
         }
     }
 
+    /// Determines the stories layout in the native collection view on iPad mode with regular horizontal size class (including split view). Number of columns for the grid layout depends on device orientation and the view's width length. If view's width is less than 800, show 2 col, otherwise show 2 col if it is in portrait and 3 col if it is in landscape mode.
+    private func collectionSectionForWideLayout(with stories: [CollectionStory], width: CGFloat, margin: CGFloat) -> NSCollectionLayoutSection {
+        if self.view.bounds.width >= Constants.minWidthBoundaryForFullColumnLayout && !UIDevice.current.orientation.isPortrait {
+            return collectionSectionForiPadLayout(with: stories, width: width, margin: margin, numberOfColumns: 3)
+        } else {
+            return collectionSectionForiPadLayout(with: stories, width: width, margin: margin, numberOfColumns: 2)
+        }
+    }
+
+    /// Determines the stories layout for the native collection view on iPad mode with regular horizontal size class
+    /// - Parameters:
+    ///   - stories: list of stories associated with collection to display
+    ///   - width: width that the section occupies
+    ///   - margin: padding adding to the side of the section
+    /// - Returns: section layout for stories on iPad view and regular horizontal size class
+    private func collectionSectionForiPadLayout(with stories: [CollectionStory], width: CGFloat, margin: CGFloat, numberOfColumns: CGFloat) -> NSCollectionLayoutSection {
+        let numberOfRows = (CGFloat(stories.count) / numberOfColumns).rounded(.up)
+
+        let recommendationsHeight: [CGFloat] = stories.map { story in
+            return RecommendationCell.fullHeight(viewModel: model.storyViewModel(for: story), availableWidth: width / numberOfColumns - (margin * 2)) + margin
```

Comment:
Maybe pull `availableWidth` into it's own function?

Reply:
I kept the code consistent with previous logic, but I think it's best to refactor, so I'll go ahead and refactor this! thanks for pointing it out!